src/Umbraco.Core/Services/Implement/ContentService.cs

```diff
@@ -900,6 +910,11 @@ public PublishResult SaveAndPublish(IContent content, string[] cultures, int use
             if (content == null) throw new ArgumentNullException(nameof(content));
             if (cultures == null) throw new ArgumentNullException(nameof(cultures));
 
+            if (content.Name.Length > 255)
+            {
+                throw new Exception("Name cannot be more than 255 characters in length.");
```

Comment:
See prev comment re: InvalidOperationException

Reply:
Made the changes as requested @Shazwazza . As I said earlier I just couldnt work out how to get it working for the scenario where I try and change the name of a node to something beyond 255 characters. But thats me I guess! If there is anything more I can do to help let me know :-) Else see you at the retreat :-)